File: JAVA_COURSE/JAVA/29_Processor/SimpleThreadManager/src/simplethreadmanager/SimpleThreadManager.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package simplethreadmanager;

/**
 *
 * @author LV
 */
public class SimpleThreadManager
{
    public static void main(String[] arg) {
        for(int i=0; i<1000; i++) {
            Thread first = new SimpleThread();
            first.start();
        }
    }
}
 
class SimpleThread extends Thread
{
    @Override
    public void run() {
        try {
            long pause = Math.round(Math.random()*7000);
            Thread.sleep(pause);
            System.out.println("Simple Thread - pause="+pause);
        } catch(InterruptedException i_ex) {
            
        }
    }
}

File: JAVA_COURSE/KOTLIN/29_Processor/SimpleThreadManager/src/Kotlinsimplethreadmanager/SimpleThreadManager.kt
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Kotlinsimplethreadmanager

/**
 *
 * @author LV
 */
object SimpleThreadManager {
    @JvmStatic
    fun main(arg: Array<String>) {
        for (i in 0..999) {
            val first = SimpleThread()
            first.start()
        }
    }
}

internal class SimpleThread : Thread() {
    override fun run() {
        try {
            val pause = Math.round(Math.random() * 7000)
            Thread.sleep(pause)
            println("Simple Thread - pause=$pause")
        } catch (i_ex: InterruptedException) {

        }

    }
}